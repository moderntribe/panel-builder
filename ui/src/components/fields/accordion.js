import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import _ from 'lodash';
import striptags from 'striptags';
import autobind from 'autobind-decorator';
import TweenMax, { Power3 } from 'gsap';

import FieldBuilder from '../shared/field-builder';

import styles from './accordion.pcss';
import * as domTools from '../../util/dom/tools';
import * as panelConditionals from '../../util/dom/panel-conditionals';
import * as events from '../../util/events';

const tw = window.TweenMax ? window.TweenMax : TweenMax;
const p3 = window.Power3 ? window.Power3 : Power3;

class Accordion extends Component {
	constructor(props) {
		super(props);
		this.state = {
			id: _.uniqueId('accordion-'),
		};
		this.active = false;
		this.animating = false;
	}

	componentDidMount() {
		panelConditionals.initConditionalFields(domTools.closest(this.tab, '[data-panel]'));
		this.mounted = true;
		document.addEventListener('modern_tribe/panel_builder/close_accordions', this.maybeCloseAccordion);
	}

	componentWillUnmount() {
		this.mounted = false;
		document.removeEventListener('modern_tribe/panel_builder/close_accordions', this.maybeCloseAccordion);
	}

	/**
	 * Gets the header which toggles the group into view
	 * @returns {XML}
	 */

	getHeader() {
		const headerClasses = classNames({
			[styles.header]: true,
			'panel-builder__row-header': true,
		});
		const arrowClasses = classNames({
			'dashicons': true,
			[styles.arrow]: true,
			'panel-builder__row-arrow': true,
			'dashicons-arrow-right-alt2': true,
		});

		return (
			<div
				className={headerClasses}
				onClick={this.handleHeaderClick}
			>
				<h3>{striptags(this.props.label)}</h3>
				<i className={arrowClasses} />
			</div>
		);
	}

	/**
	 * Gets the fields, called when state is active.
	 * @returns {XML}
	 */

	@autobind
	getFields() {
		const fieldClasses = classNames({
			[styles.fields]: true,
			'panel-builder__accordion-fields': true,
		});
		const descriptionClasses = classNames({
			[styles.description]: true,
			'panel-builder__field-description': true,
		});
		const parentMap = this.props.parentMap.slice();
		parentMap.push(this.props.name);

		return (
			<div ref={r => this.fields = r} className={fieldClasses}>
				<div className={styles.fieldWrap}>
					<p className={descriptionClasses}>{this.props.description}</p>
					<FieldBuilder
						fields={this.props.fields}
						data={this.props.data}
						depth={this.props.depth}
						parent={this.props.name}
						parentMap={parentMap}
						parentType={this.props.name}
						index={this.props.panelIndex}
						indexMap={this.props.indexMap}
						type={this.props.type}
						updatePanelData={this.props.updatePanelData}
					/>
				</div>
			</div>
		);
	}

	maybeAnimateFields() {
		this.animating = true;
		if (!this.active) {
			this.el.classList.remove(styles.active);
			this.fields.classList.remove(styles.animated);
			tw.to(this.fields, 0.6, { ease: p3.easeOut, height: 0 });
			_.delay(() => { this.animating = false; }, 600);
			return;
		}
		this.el.classList.add(styles.active);
		tw.set(this.fields, { height: 'auto' });
		tw.from(this.fields, 0.6, { ease: p3.easeOut, height: 0 });
		_.delay(() => {
			this.animating = false;
			this.fields.classList.add(styles.animated);
		}, 600);
		events.trigger({ event: 'modern_tribe/panel_builder/close_accordions', native: false, data: { id: this.state.id } });
	}

	/**
	 * Handles the header click and toggles fields into/out of view.
	 */

	@autobind
	handleHeaderClick() {
		if (this.animating) {
			return;
		}
		this.active = !this.active;
		this.maybeAnimateFields();
	}

	/**
	 * Listen to level 1 accordion open events. If the id does not match the one that triggered the open event,
	 * close this instance
	 *
	 * @param e Contains the id of the one that triggered the open event
	 */

	@autobind
	maybeCloseAccordion(e) {
		if (!this.mounted) {
			return;
		}
		if (this.state.id === e.detail.id) {
			return;
		}
		if (!this.active) {
			return;
		}

		this.handleHeaderClick();
	}

	render() {
		const fieldClasses = classNames({
			[styles.field]: true,
			'panel-builder__field': true,
			'panel-builder__accordion-field': true,
		});

		return (
			<div ref={r => this.el = r} className={fieldClasses}>
				{this.getHeader()}
				{this.getFields()}
			</div>
		);
	}
}

Accordion.propTypes = {
	data: PropTypes.object,
	default: PropTypes.array,
	depth: PropTypes.number,
	description: PropTypes.string,
	fields: PropTypes.array,
	handleExpanderClick: PropTypes.func,
	hidePanel: PropTypes.func,
	indexMap: PropTypes.array,
	parentMap: PropTypes.array,
	label: PropTypes.string,
	liveEdit: PropTypes.bool,
	name: PropTypes.string,
	nestedGroupActive: PropTypes.func,
	panelIndex: PropTypes.number,
	panelLabel: PropTypes.string,
	parent: PropTypes.string,
	parentIndex: PropTypes.number,
	type: PropTypes.string,
	updatePanelData: PropTypes.func,
};

Accordion.defaultProps = {
	data: {},
	default: [],
	depth: 0,
	description: '',
	fields: [],
	handleExpanderClick: () => {},
	hidePanel: () => {},
	indexMap: [],
	parentMap: [],
	label: '',
	liveEdit: false,
	name: '',
	nestedGroupActive: () => {},
	panelIndex: 0,
	panelLabel: '',
	parent: '',
	parentIndex: 0,
	type: '',
	updatePanelData: () => {},
};

export default Accordion;
