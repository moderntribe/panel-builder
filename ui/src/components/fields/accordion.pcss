@import '../../shared/spacing.pcss';
@import '../../shared/colors.pcss';
@import '../../shared/transitions.pcss';

.field {
	border-bottom: 1px solid var(--border-color);
	composes: fieldContainer from "../../shared/fields.pcss";
	margin: 0 -15px;
	padding: 0 10px;
	width: calc(100% + 30px);
}

.label {
	composes: inputLabel from "../../shared/fields.pcss";
}

.arrow {
	composes: arrow from "../../shared/accordion.pcss";
	transition: var(--transition-accordion);
	left: 0;
	top: 24px;

	&:before {
		content: "\f139";
		font-family: 'dashicons';
		color: #4A4A4A;
	}

	[data-live-edit="false"] & {
		transform: rotate(0deg);
	}

	.active & {
		transform: rotate(90deg);
	}
}

.header {
	composes: header from "../../shared/accordion.pcss";
	margin: 0 auto;
	background: none;
	padding: 25px 0;
	border: 0;

	h3 {
		margin-left: 25px;
	}

	.description {
		margin-bottom: 0;
	}
}

.back {
	composes: back from "../../shared/accordion.pcss";
}

.action {
	composes: action from "../../shared/accordion.pcss";
}

.backButton {
	composes: backButton from "../../shared/accordion.pcss";
}

.fields {
	padding: 0;
	height: 0;
	overflow: hidden;

	&.animated {
		overflow: visible;
	}

	.fieldWrap {
		[data-settings="false"] {
			display: block;
		}
	}
}

.fieldWrap {
	overflow-y: visible;
	height: auto;
	padding-bottom: 25px;
	padding-left: 25px;
	width: calc(100% - 25px);
}

.description {
	composes: inputDescription from "../../shared/fields.pcss";
	padding: 0 20px 10px 0;
	margin-top: 0;
}
