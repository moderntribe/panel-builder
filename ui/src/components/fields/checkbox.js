import React, { Component } from 'react';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';
import _ from 'lodash';
import classNames from 'classnames';

import LabelTooltip from './partials/label-tooltip';
import styles from './checkbox.pcss';
import * as styleUtil from '../../util/dom/styles';

class Checkbox extends Component {
	state = {
		data: this.props.data,
	};

	@autobind
	handleChange(e) {
		const key = e.currentTarget.value;
		const data = _.cloneDeep(this.state.data);
		data[key] = this.state.data[key] ? 0 : 1;
		this.setState({
			data,
		});
		this.props.updatePanelData({
			depth: this.props.depth,
			indexMap: this.props.indexMap,
			parentMap: this.props.parentMap,
			name: this.props.name,
			value: data,
		});
	}

	render() {
		const labelClasses = classNames({
			[styles.label]: true,
			'panel-field-label': true,
		});
		const fieldClasses = classNames({
			[styles.field]: true,
			[styles.vertical]: this.props.layout === 'vertical',
			[styles.horizontal]: this.props.layout === 'horizontal',
			'panel-field': true,
			'panel-conditional-field': true,
		});
		const itemStyles = this.props.layout === 'horizontal' && styleUtil.optionStyles(this.props) || {};
		const Options = _.map(this.props.options, option => (
			<li
				key={_.uniqueId('checkbox-id-')}
				className={styles.item}
				style={itemStyles}
			>
				<label className={styles.checkboxLabel}>
					<input
						type="checkbox"
						name={`modular-content-${this.props.name}[]`}
						value={option.value}
						tabIndex={0}
						className={styles.checkbox}
						onChange={this.handleChange}
						checked={this.state.data && this.state.data[option.value]} // eslint-disable-line
						data-option-type="multiple"
						data-field="checkbox"
						data-depth={this.props.depth}
					/>
					<span />
					{option.label}
				</label>
			</li>
		));

		return (
			<div className={fieldClasses}>
				<label className={labelClasses}>
					{this.props.label}
					{this.props.description.length ? <LabelTooltip content={this.props.description} /> : null}
				</label>
				<ul className={styles.list}>
					{Options}
				</ul>
			</div>
		);
	}
}

Checkbox.propTypes = {
	label: PropTypes.string,
	name: PropTypes.string,
	depth: PropTypes.number,
	indexMap: PropTypes.array,
	parentMap: PropTypes.array,
	description: PropTypes.string,
	strings: PropTypes.object,
	default: PropTypes.object,
	options: PropTypes.array,
	data: PropTypes.object,
	panelIndex: PropTypes.number,
	layout: PropTypes.string,
	option_width: PropTypes.number,
	updatePanelData: PropTypes.func,
};

Checkbox.defaultProps = {
	label: '',
	name: '',
	description: '',
	indexMap: [],
	parentMap: [],
	depth: 0,
	strings: {},
	default: {},
	options: [],
	data: {},
	panelIndex: 0,
	layout: 'vertical',
	option_width: 4,
	updatePanelData: () => {},
};

export default Checkbox;
