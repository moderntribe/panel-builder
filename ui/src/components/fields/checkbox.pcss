@import '../../shared/spacing.pcss';
@import '../../shared/colors.pcss';

.field{
	composes: fieldContainer from "../../shared/fields.pcss";
	max-width: 340px;
}

.checkbox {
	composes: checkbox from "../../shared/inputs.pcss";
	vertical-align: middle;
}

.vertical {

}

.horizontal {

	.list {
		display: flex;
		flex-direction: row;
		flex-wrap: wrap;

		.item {
			/* list item */

			.checkboxLabel {
				padding-right: 10px;
			}
		}
	}
}

.checkboxLabel {
	composes: label from "../../shared/inputs.pcss";
	font-size: 11px;
	height: 25px;
	line-height: 27px;
}

.list {
	composes: list from "../../shared/fields.pcss";

	li {
		margin-bottom: 0;
	}
}

.label {
	composes: inputLabel from "../../shared/fields.pcss";
}

.description {
	composes: inputDescription from "../../shared/fields.pcss";
}
