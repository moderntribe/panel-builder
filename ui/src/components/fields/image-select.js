import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';
import classNames from 'classnames';
import _ from 'lodash';

import styles from './image-select.pcss';
import { trigger } from '../../util/events';
import * as EVENTS from '../../constants/events';
import { UI_I18N } from '../../globals/i18n';
import * as styleUtil from '../../util/dom/styles';
import LabelTooltip from './partials/label-tooltip';

export class ImageSelect extends Component {
	constructor(props) {
		super(props);
		this.state = {
			id: _.uniqueId('image-select-'),
			hidden: this.props.can_add_columns,
			layoutRequest: null,
			value: this.props.data,
		};
		this.confirmColumnInjection = this.confirmColumnInjection.bind(this);
	}

	componentDidMount() {
		document.addEventListener(EVENTS.INJECT_LAYOUT, this.confirmColumnInjection);
	}

	componentWillUnmount() {
		document.removeEventListener(EVENTS.INJECT_LAYOUT, this.confirmColumnInjection);
	}

	@autobind
	handleHeader() {
		if (!this.props.can_add_columns) {
			return;
		}
		this.setState({ hidden: !this.state.hidden });
	}

	confirmColumnInjection(e) {
		if (e.detail.id !== this.state.id) {
			return;
		}
		this.updateState(this.state.layoutRequest);
	}

	updateState(value) {
		if (this.props.can_add_columns) {
			this.props.injectColumns({
				indexMap: this.props.indexMap,
				value,
			});
		}
		this.setState({
			value,
			hidden: this.props.can_add_columns,
		}, () => {
			if (!this.props.can_add_columns) {
				return;
			}
			trigger({
				event: EVENTS.COLUMNS_UPDATED,
				native: false,
			});
		});
		this.props.updatePanelData({
			depth: this.props.depth,
			indexMap: this.props.indexMap,
			parentMap: this.props.parentMap,
			name: this.props.name,
			value,
		});
	}

	@autobind
	handleChange(e) {
		const value = e.currentTarget.value;
		if (this.props.can_add_columns) {
			trigger({
				event: EVENTS.OPEN_DIALOG,
				native: false,
				data: {
					type: 'error',
					confirm: true,
					heading: UI_I18N['message.confirm_layout'],
					data: {
						id: this.state.id,
					},
					confirmCallback: EVENTS.INJECT_LAYOUT,
				},
			});
			this.setState({ layoutRequest: value });
			return;
		}
		this.updateState(value);
	}

	render() {
		const imgSelectLabelClasses = classNames({
			'plimageselect-label': true,
			[styles.imageSelectLabel]: true,
			[styles.isColumnImg]: this.props.can_add_columns,
		});
		const itemStyles = styleUtil.optionStyles(this.props);

		const Options = _.map(this.props.options, option => (
			<label
				className={imgSelectLabelClasses}
				key={_.uniqueId('option-img-sel-id-')}
				style={itemStyles}
			>
				<input
					type="radio"
					name={`modular-content-${this.props.name}`}
					value={option.value}
					onChange={this.handleChange}
					checked={this.state.value === option.value}
					data-option-type="single"
					data-field="image-select"
					data-depth={this.props.depth}
				/>
				<div
					className={styles.optionImage}
				>
					<span className={styles.optInner}>
						<img src={option.src} alt={option.label} />
					</span>
				</div>
				{!this.props.can_add_columns && option.label}
			</label>
		));

		const labelClasses = classNames({
			[styles.label]: true,
			'panel-field-label': true,
		});

		const arrowClasses = classNames({
			'dashicons': true,
			'dashicons-arrow-right-alt2': true,
			[styles.arrow]: this.props.can_add_columns,
			[styles.arrowUp]: this.state.hidden,
		});

		const containerClasses = classNames({
			[styles.container]: true,
			[styles.hidden]: this.state.hidden,
			[styles.isColumn]: this.props.can_add_columns,
		});
		const fieldClasses = classNames({
			[styles.field]: true,
			[styles.hidden]: this.state.hidden,
			[styles.isColumnType]: this.props.can_add_columns,
			'panel-field': true,
			'panel-conditional-field': true,
		});

		return (
			<div className={fieldClasses}>
				<label className={labelClasses} onClick={this.handleHeader}>
					{this.props.label}
					{!this.props.can_add_columns && this.props.description.length ? <LabelTooltip content={this.props.description} /> : null}
					{this.props.can_add_columns && <i className={arrowClasses} />}
				</label>
				<div className={containerClasses}>
					{Options}
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	injectColumns: data => dispatch(injectColumns(data)),
});

ImageSelect.propTypes = {
	can_add_columns: PropTypes.bool,
	label: PropTypes.string,
	name: PropTypes.string,
	description: PropTypes.string,
	strings: PropTypes.object,
	indexMap: PropTypes.array,
	parentMap: PropTypes.array,
	depth: PropTypes.number,
	default: PropTypes.string,
	option_width: PropTypes.number,
	options: PropTypes.array,
	data: PropTypes.string,
	panelIndex: PropTypes.number,
	injectColumns: PropTypes.func,
	updatePanelData: PropTypes.func,
};

ImageSelect.defaultProps = {
	can_add_columns: false,
	label: '',
	name: '',
	description: '',
	indexMap: [],
	parentMap: [],
	strings: {},
	depth: 0,
	default: '',
	option_width: 6,
	options: [],
	data: '',
	panelIndex: 0,
	injectColumns: () => {},
	updatePanelData: () => {},
};

export default connect(null, mapDispatchToProps)(ImageSelect);
