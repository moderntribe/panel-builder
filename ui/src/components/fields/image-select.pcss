@import '../../shared/spacing.pcss';
@import '../../shared/colors.pcss';
@import '../../shared/mq.pcss';
@import '../../shared/transitions.pcss';

.field {
	composes: fieldContainer from "../../shared/fields.pcss";
	transform: translate3d(0, 0, 0);
}

.label {
	composes: inputLabel from "../../shared/fields.pcss";
	display: block;
}

.arrow {
	transition: transform 200ms ease-in-out;
	transform: rotate(-90deg);
	margin-left: 6px;

	&.arrowUp {
		transform: rotate(90deg);
	}
}

.description {
	composes: inputDescription from "../../shared/fields.pcss";

	.isColumnType & {
		transition: color 600ms ease-in-out;
		color: var(--dialog-error);
	}

	.hidden.isColumnType & {
		color: var(--description-text-color);
	}
}

.optionImage {
	width: 100%;

	img {
		background: var(--button-primary-bg-color);
		opacity: 0.4;
		transition: all 0.1s ease-out;
		-webkit-backface-visibility: hidden;
		backface-visibility: hidden;
		display: block;
		width: 100%;
		height: auto;
		box-sizing: border-box;
		margin-bottom: 5px;

		.isColumnImg & {
			padding: 25px 5px;
		}
	}
}

.container {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	overflow: hidden;
	transition: max-height 600ms ease-in-out;

	&.isColumn {
		max-height: 1000px;
	}

	&.hidden {
		max-height: 0;
	}
}

[data-live-edit="false"] {

	.imageSelectLabel {
		font-size: 12px;
		line-height: 1.4;
		margin-bottom: 12px;
		min-width: 50%;
		max-width: 50%;

		@media (--viewport-small) {
			min-width: auto !important;
			max-width: none !important;
		}

		@media (--viewport-large) {
			font-size: 11px;
			margin-right: 5px;
			max-width: calc(12.5% - 5px) !important;
		}
	}
}

[data-live-edit="true"] {

	.imageSelectLabel {
		font-size: 10px;
		line-height: 1.4;
		margin-bottom: 16px;
		max-width: none;
	}

	[data-expanded="true"] {

		.imageSelectLabel {
			font-size: 12px;
		}
	}
}

.imageSelectLabel {
	color: var(--compact-label);
	display: inline-block;
	vertical-align: top;
	text-align: center;
	font-weight: bold;
	box-sizing: border-box;
	position: relative;
	margin-bottom: 5px;

	input[type="radio"] {
		display: none;
	}

	.optionImage .optInner {
		display: block;
		border: 2px solid transparent;

		img {
			border: 1px solid var(--border-input);
			border-radius: 2px;
			display: block;
			margin: 0;
			pointer-events: none;
		}
	}

	.optionImage .optInner:hover img {
		border: 1px solid var(--button-new-blue);
		opacity: 0.65;
	}

	input[type="radio"]:checked + .optionImage {

		img {
			background: var(--color-white);
			border-color: var(--button-new-blue);
			opacity: 1;
		}
	}
}
