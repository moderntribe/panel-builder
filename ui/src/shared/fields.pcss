@import '../shared/colors.pcss';

.inputLabel {
	cursor: auto;
	display: block;
	color: var(--panel-label-color) ;
	font-style: normal;
	font-weight: 600;
	font-size: 13px;
	margin-bottom: 8px;
	position: relative;
	transform: translate3d(0, 0, 0);
	max-width: 225px;
}

.inputDescription {
	display: none;
	letter-spacing: .05em;
	margin-top: 8px;
	font-style: normal;
	font-size: 85%;
	color: var(--description-text-color) ;
	clear: both;
	transform: translate3d(0, 0, 0);
}

.list {
	position: relative;
	padding: 1px;
	margin: 0;

	li {
		display: block;
		margin-bottom: 5px;
		padding: 0;
		float: none;
	}
}

.fieldContainer {
	margin-bottom: 18px;
	clear: both;
}

.standardLink {
	color: var(--link-color);
	padding: 10px 12px;
}

.remove {
	float: right;
	color: var(--link-color);
	cursor: pointer;

	&:hover {
		color: var(--move-hover-color);
	}
}