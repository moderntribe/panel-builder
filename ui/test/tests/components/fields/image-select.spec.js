/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import { ImageSelect } from 'components/fields/image-select';
import renderer from 'react-test-renderer';

// test data
const OPTIONS1 = [
	{
		src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
		label: 'Standard',
		value: 'standard',
	},
	{
		src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
		label: 'Cards',
		value: 'cards',
	},
	{
		src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
		label: 'Full',
		value: 'full',
	},
];
const OPTIONS2 = [
	{
		src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
		label: 'Cows',
		value: 'cows',
	},
	{
		src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==',
		label: 'Orangutan',
		value: 'orangutan',
	},
];

describe('ImageSelect field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<ImageSelect
					description="This is a test description"
					label="Some test label"
					options={OPTIONS1}
					name="layout"
					default="cards"
				/>,
			).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<ImageSelect />);
		expect(wrapper.type()).toEqual('div');
	});


	it('has no image select radio buttons without properties.', () => {
		const wrapper = shallow(<ImageSelect />);
		expect(wrapper.find('input').length).toEqual(0);
	});

	it('has a label', () => {
		const wrapper = shallow(<ImageSelect />);
		expect(wrapper.find('label').length).toEqual(1);
	});

	it('allows us to set and edit props', () => {
		const wrapper = mount(
			<ImageSelect
				description="This is a test description"
				label="Some test label"
				options={OPTIONS1}
				name="layout"
				default="cards"
			/>
		);

		expect(wrapper.props().description).toEqual('This is a test description');
		wrapper.setProps({ description: 'Change description for test' });
		expect(wrapper.props().description).toEqual('Change description for test');

		expect(wrapper.props().label).toEqual('Some test label');
		wrapper.setProps({ label: 'Change label for test' });
		expect(wrapper.props().label).toEqual('Change label for test');

		expect(wrapper.props().name).toEqual('layout');
		wrapper.setProps({ name: 'animal' });
		expect(wrapper.props().name).toEqual('animal');

		expect(wrapper.props().default).toEqual('cards');
		wrapper.setProps({ default: 'cows' });
		expect(wrapper.props().default).toEqual('cows');

		expect(wrapper.props().options).toEqual(OPTIONS1);
		wrapper.setProps({ options: OPTIONS2 });
		expect(wrapper.props().options).toEqual(OPTIONS2);
	});

	it('allows us to properly formatted image select options', () => {
		const wrapper = mount(
			<ImageSelect
				description="This is a test description"
				label="Some test label"
				options={OPTIONS1}
				name="layout"
				default="cards"
			/>
		);

		expect(wrapper.find('label.plimageselect-label').length).toEqual(3);
		expect(wrapper.find('label.plimageselect-label').first().find('input').length).toEqual(1);
		expect(wrapper.find('label.plimageselect-label').first().find('img').length).toEqual(1);
		expect(wrapper.find('label.plimageselect-label').first().find('input').props().value).toEqual('standard');
		expect(wrapper.find('label.plimageselect-label').first().text()).toEqual('Standard');
	});
});
