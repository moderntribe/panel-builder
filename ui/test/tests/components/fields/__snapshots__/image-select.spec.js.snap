exports[`ImageSelect field renders correctly 1`] = `
<div
  className="field panel-field panel-conditional-field">
  <label
    className="label panel-field-label"
    onClick={[Function]}>
    Some test label
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <div
    className="container">
    <label
      className="plimageselect-label imageSelectLabel"
      style={
        Object {
          "width": "50%",
        }
      }>
      <input
        checked={false}
        data-depth={0}
        data-field="image-select"
        data-option-type="single"
        name="modular-content-layout"
        onChange={[Function]}
        type="radio"
        value="standard" />
      <div
        className="optionImage">
        <span
          className="optInner">
          <img
            alt="Standard"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==" />
        </span>
      </div>
      Standard
    </label>
    <label
      className="plimageselect-label imageSelectLabel"
      style={
        Object {
          "width": "50%",
        }
      }>
      <input
        checked={false}
        data-depth={0}
        data-field="image-select"
        data-option-type="single"
        name="modular-content-layout"
        onChange={[Function]}
        type="radio"
        value="cards" />
      <div
        className="optionImage">
        <span
          className="optInner">
          <img
            alt="Cards"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==" />
        </span>
      </div>
      Cards
    </label>
    <label
      className="plimageselect-label imageSelectLabel"
      style={
        Object {
          "width": "50%",
        }
      }>
      <input
        checked={false}
        data-depth={0}
        data-field="image-select"
        data-option-type="single"
        name="modular-content-layout"
        onChange={[Function]}
        type="radio"
        value="full" />
      <div
        className="optionImage">
        <span
          className="optInner">
          <img
            alt="Full"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z/C/HgAGgwJ/lK3Q6wAAAABJRU5ErkJggg==" />
        </span>
      </div>
      Full
    </label>
  </div>
</div>
`;
